python_docs_theme/static/copybutton.js

```diff
@@ -1,64 +1,63 @@
-        var jthis = $(this);
-        if (jthis.find('.gp').length > 0) {
-            var button = $('<span class="copybutton">&gt;&gt;&gt;</span>');
-            button.css(button_styles)
-            button.attr('title', hide_text);
-            button.data('hidden', 'false');
-            jthis.prepend(button);
-        }
-        // tracebacks (.gt) contain bare text elements that need to be
-        // wrapped in a span to work with .nextUntil() (see later)
-        jthis.find('pre:has(.gt)').contents().filter(function() {
-            return ((this.nodeType == 3) && (this.data.trim().length > 0));
-        }).wrap('<span>');
-    });
+    const hide_text = 'Hide the prompts and output';
+    const show_text = 'Show the prompts and output';
 
-    // define the behavior of the button when it's clicked
-    $('.copybutton').click(function(e){
-        e.preventDefault();
-        var button = $(this);
-        if (button.data('hidden') === 'false') {
+    const button = document.createElement("span")
+    button.classList.add("copybutton")
+    button.innerText = ">>>"
+    button.title = hide_text
+    button.dataset.hidden = false
+    const buttonClick = event => {
+        // define the behavior of the button when it's clicked
+        event.preventDefault();
+        const buttonEl = event.currentTarget
+        const codeEl = buttonEl.nextElementSibling
+        if (buttonEl.dataset.hidden === 'false') {
             // hide the code output
-            button.parent().find('.go, .gp, .gt').hide();
-            button.next('pre').find('.gt').nextUntil('.gp, .go').css('visibility', 'hidden');
-            button.css('text-decoration', 'line-through');
-            button.attr('title', show_text);
-            button.data('hidden', 'true');
+            codeEl.querySelectorAll('.go, .gp, .gt').forEach(el => el.hidden = true)
+            codeEl.querySelectorAll('.gt').forEach(el => {
+                while ((el = el.nextSibling) && el.nodeType !== Node.DOCUMENT_NODE) {
+                    if (el.nodeType === Node.ELEMENT_NODE && el.matches(".gp, .go")) break;
+                    if (el.nodeType === Node.TEXT_NODE && el.textContent.trim()) {
+                        const wrapper = document.createElement('span');
+                        el.after(wrapper);
+                        wrapper.appendChild(el);
+                        el = wrapper
+                    }
+                    el.hidden = true
+                }
+            })
+            buttonEl.title = show_text
+            buttonEl.dataset.hidden = "true"
         } else {
             // show the code output
-            button.parent().find('.go, .gp, .gt').show();
-            button.next('pre').find('.gt').nextUntil('.gp, .go').css('visibility', 'visible');
-            button.css('text-decoration', 'none');
-            button.attr('title', hide_text);
-            button.data('hidden', 'false');
+            codeEl.childNodes.forEach(el => el.hidden = false)
```

Comment:
IIUC here you are just assuming that everything inside the code block was already visible by default, and you are setting `.hidden = false` to each element.  Do you think this might conflict with other things (e.g. Sphinx extensions) that might introduce hidden elements?

Reply:
Yes, this replicates the current behaviour -- I would suggest merging this as-is to keep the behaviour consistent and if the unconditional hiding is a problem we can change it later.

A

Reply:
The original implementation was using `button.parent().find('.go, .gp, .gt')` `button.next('pre').find('.gt').nextUntil('.gp, .go')` to select the elements, but in this version it seems you are doing this only in the `if` branch but not in the `else`.

Also, without JQuery, selecting the tracebacks seems more convoluted, so I understand you might not want to duplicate it in the `else`, but perhaps you could create a function to show/hide tracebacks and call it in both the `if` and the `else`?  This will make the implementation closer to the original and also more readable/understandable.

Reply:
I've added a new `getHideableCopyButtonElements` generator function.

A